### ISSUE TITLE
Unencrypted credentials in helm values

### ISSUE THREAD

**Issue by timbrd** [fjogeleit/policy-reporter#184]
Hi.

I am currently trying to configure email reporting. Unfortunately I could only find a way to specify the mailserver credentials in the values.yaml file. The file is going to be pushed to my gitops repository. Since the credentials are unencrypted in the values.yaml file, they are exposed to everyone who is able to access my git repository.

I would rather like to create a secret containing the mailserver credentials beforehand and just reference it in the values.yaml file.

Could you implement a way to reference a secret in the values.yaml file?

Thanks

**Comment by fjogeleit** [comment]
Hey, you can also create your own configuration secret and configure it in the values.yaml:

https://github.com/kyverno/policy-reporter/blob/main/charts/policy-reporter/values.yaml#L220

does this help or how would your workflow looks like?

**Comment by fjogeleit** [comment]
Oh sorry, I see this is only used for the deployment, not the email reports. So you would like to use a similar approach for the email reports, am I right?

**Comment by timbrd** [comment]
> Oh sorry, I see this is only used for the deployment, not the email reports. So you would like to use a similar approach for the email reports, am I right?

Yes, I would like to create the secret with the smtp configuration by myself, which preferebly only contains the smtp username and password. The other smtp configuration should be specified in the helm chart.
Check the [values.yaml](https://github.com/grafana/helm-charts/blob/main/charts/grafana/values.yaml#L652) of the grafana helm chart as an example.

**Comment by fjogeleit** [comment]
okay, thanks. I will have a look on it.

**Comment by fjogeleit** [comment]
I created a PR to manage the SMTP configuration complete or partially as secret.

You can configure each key that is also available in the helm chart over the secret: host, port, encryption, from, username, password

The secret configuration has a higher priority as the configuration in the release values, when both are provided. Would this work for you?

**Comment by timbrd** [comment]
> I created a PR to manage the SMTP configuration complete or partially as secret.
> 
> You can configure each key that is also available in the helm chart over the secret: host, port, encryption, from, username, password
> 
> The secret configuration has a higher priority as the configuration in the release values, when both are provided. Would this work for you?

Looks good to me. I couldn't test it yet though, since the new image hasn't been released yet.

**Comment by fjogeleit** [comment]
If you want to test it you could use

```yaml
image:
  tag: dev
  registry: docker.io
  repository: fjogeleit/policy-reporter
```

with the PR changes. I tested it with this build.

I will release the new version tomorrow.

**Comment by timbrd** [comment]
Using the dev image I was able to verify that my use case works.

Thanks, can't wait for the release. :)

**Comment by fjogeleit** [comment]
Thanks for the fast verification. So I decided to release it already today. It is released as Helm Chart Version  `2.11.3`.

I hope the email reports feature offers everything you expect. I am happy about all kind of feedback.

I close the issue, just let me know if you encounter any other problems.

**Comment by timbrd** [comment]
Thank you for the very fast implementation. I just updated policy reporter to the new version on my kubernetes cluster.
